### PR TITLE
Move VSCode diagnostics workaround into client code

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

File: crates/rust-analyzer/src/main_loop.rs [hunk 1]
```diff
@@ -327,29 +327,15 @@ impl GlobalState {
                     continue;
                 }
 
-                let url = file_id_to_url(&self.vfs.read().0, file_id);
-                let mut diagnostics =
+                let uri = file_id_to_url(&self.vfs.read().0, file_id);
+                let diagnostics =
                     self.diagnostics.diagnostics_for(file_id).cloned().collect::<Vec<_>>();
-                for d in &mut diagnostics {
```
Comment (lnicola) [inline crates/rust-analyzer/src/main_loop.rs:333]: There's one more in `publish_diagnostics`.

**Comment by lnicola** [comment]
Do we unflatten the messages in `related_information`? Because I'm not sure they're handled on the client.

**Comment by Veykril** [comment]
uuh, good question let me check

**Pull request merged**